File: 0239-sliding-window-maximum/0239-sliding-window-maximum.py
class Solution(object):
    def maxSlidingWindow(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        result =[]
        q=[]
        
        for r in range(len(nums)):
            while q and nums[r] > q[-1][1]:
                q.pop()
            while q and r-q[0][0] +1 >k:
                q.pop(0)
            q.append((r,nums[r]))
            if r >=k-1:
                result.append(q[0][1])
        return result





        

File: 0239-sliding-window-maximum/0239-sliding-window-maximum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var maxSlidingWindow = function(nums, k) {
    let r = 0;
    let result = [];
    let deque=[];
    let p=0;
    for(r=0;r<nums.length;r++){
        while( deque.length && nums[deque[deque.length -1]] < nums[r]) {deque.pop();}
        while( deque.length && r-deque[0]+1 >k) deque.shift();
        deque.push(r);
        if(r>=k-1)
        result.push(nums[deque[0]]);
        
    }
    return result;

    
};